### PR TITLE
Add diff

### PR DESCRIPTION
**Pull request by Badgie**
Implements behaviour to check for diffs in the data cf. the previous data from the database. The diffs are then sent to the relevant endpoint in the API, or written to separate files.

The diffs are defined as:
- `insert`: the set difference between the new and old data
- `delete`: the set difference between the old and new data
- `update`: any row with **any** changed field in the set intersection between the new and old data

### CODE REVIEW

**Review by joandrsn — APPROVED** [review]

File: filter_xml/data_processor.py [hunk 1]
```diff
@@ -154,3 +155,47 @@ def valid_production_unit(self, restaurant: dict) -> bool:
...
         """
         return self._has_pnr(restaurant) and self._has_cvr(restaurant)
+
+    def create_diff(self, res: list) -> tuple:
+        current_db = self._outputter.get()
+        current_ids = {x['name_seq_nr'] for x in current_db}
+        res_ids = {x['name_seq_nr'] for x in res}
+
+        res_by_key = {x['name_seq_nr']: x for x in res}
+        current_by_key = {x['name_seq_nr']: x for x in current_db}
```
Comment (NicklasHugoy) [inline filter_xml/data_processor.py:171]: Is it possible to use only `res_by_key` and `current_by_key` to avoid constructing all 4 data structures?

Comment (NicklasHugoy) [reply filter_xml/data_processor.py:171]: and then use `.keys()`

File: filter_xml/data_processor.py [hunk 2]
```diff
@@ -154,3 +155,47 @@ def valid_production_unit(self, restaurant: dict) -> bool:
...
+        insert_ids = res_ids.difference(current_ids)
+        delete_ids = current_ids.difference(res_ids)
+        update_ids = res_ids.intersection(current_ids)
+
+        insert = [res_by_key[x] for x in insert_ids]
+        update = self.check_rows_for_updates(update_ids, res_by_key, current_by_key)
+
+        return insert, update, list(delete_ids)
+
+    def check_rows_for_updates(self, ids: set, new: dict, old: dict):
```
Comment (NicklasHugoy) [inline filter_xml/data_processor.py]: Missing return type hint

File: filter_xml/data_processor.py [hunk 3]
```diff
@@ -154,3 +155,47 @@ def valid_production_unit(self, restaurant: dict) -> bool:
...
+        out = []
+
+        for id_ in ids:
+            if self.has_update(new[id_], old[id_]):
+                out.append(new[id_].copy())
+
+        return out
+
+    @staticmethod
+    def has_update(new: dict, old: dict):
```
Comment (NicklasHugoy) [inline filter_xml/data_processor.py]: Here too

**Review by NicklasHugoy — APPROVED** [review]
Needs some documentation for the methods

File: filter_xml/data_processor.py (hunk 1 above)
Comment (Badgie) [reply filter_xml/data_processor.py:171]: It is, but complexity-wise it would be the same.

File: filter_xml/data_processor.py (hunk 3 above)
Comment (Badgie) [reply filter_xml/data_processor.py]: 71674e9

File: filter_xml/data_processor.py (hunk 2 above)
Comment (Badgie) [reply filter_xml/data_processor.py]: 664c823

**Review by NicklasHugoy — APPROVED** [review]

**Pull request closed without merging**